src/contexts/ListsContext.tsx

```diff
@@ -1,92 +1,140 @@
-import { createContext, useContext, useState } from "react";
+import { createContext, useContext, useEffect, useState } from "react";
 import { List } from "../types/List";
 import { Task } from "../types/Task";
-import { DefaultLists } from "../data/tasks";
+import { Id } from "../types/Id";
 
 interface ListsContextProps {
+  currentListId: Id | undefined;
+  updateCurrentListId: (id: Id | undefined) => void;
   addList: (list: List) => void;
-  addTask: (listId: number, task: Task) => void;
-  deleteList: (id: number) => void;
-  deleteTask: (listId: number, taskId: number) => void;
+  addTask: (listId: Id, task: Task) => void;
+  deleteList: (id: Id) => void;
+  deleteTask: (listId: Id, taskId: Id) => void;
   lists: List[];
-  updateList: (id: number, updatedList: Partial<List>) => void;
-  updateTask: (
-    listId: number,
-    taskId: number,
-    updatedTask: Partial<Task>,
-  ) => void;
+  updateList: (updatedList: Partial<List>) => void;
+  updateTask: (listId: Id, updatedTask: Partial<Task>) => void;
+  getListByListId: (listId: Id) => List | undefined;
 }
 
 const ListsContext = createContext<ListsContextProps | undefined>(undefined);
 
 export const ListsProvider = ({ children }: { children: React.ReactNode }) => {
-  const [lists, setLists] = useState<List[]>(DefaultLists);
+  const [lists, setLists] = useState<List[]>([]);
+
+  const [currentListId, setCurrentListId] = useState<Id | undefined>(
+    lists?.[0]?.id
+  );
+
+  useEffect(() => {
+    const storedLists = localStorage.getItem("lists");
+    const updatedList = storedLists ? JSON.parse(storedLists) : [];
+
+    setLists(updatedList);
+    setCurrentListId(updatedList.length ? updatedList[0].id : null);
+  }, []);
+
+  const saveToLocalStorage = (updatedLists: List[]) => {
+    localStorage.setItem("lists", JSON.stringify(updatedLists));
+  };
+
+  const updateCurrentListId = (id: Id | undefined) => {
+    setCurrentListId(id);
+  };
 
   const addList = (list: List) => {
-    setLists((prevLists) => [...prevLists, { ...list, tasks: [] }]);
+    setLists((prevLists) => {
+      const updatedLists = [{ ...list, tasks: [] }, ...prevLists];
```

Comment:
Do not perform all these operations inside set states. It should simply be `setLists(updatedLists)` and rest of operations should be outside the function. Please fix this in the entire file.